builtin/logical/aws/secret_access_keys.go

```diff
@@ -69,11 +68,10 @@ func genUsername(displayName, policyName, userType, usernameTemplate string) (re
 		// To prevent template from exceeding IAM length limits
 		if len(ret) > 64 {
 			ret = ret[0:64]
-			warning = "the calling token display name/IAM policy name were truncated to 64 characters to fit within IAM username length limits"
+			warning = fmt.Sprintf("the calling token's %s user name was truncated to 64 characters to fit within username length limits", userType)
```

Comment:
I like Michael's idea of returning an error if custom username template results in usernames that's longer than the limit so perhaps we can turn this into an error (you'll also have to update the tests).

Reply:
Updated